pkg/kubelet/kubelet_server_journal.go

```diff
@@ -148,13 +148,64 @@ func (a *journalArgs) Copy(w io.Writer) {
 	}
 }
 
-// copyForBoot invokes journalctl or Get-WinEvent with the  provided args
+// copyForBoot invokes the OS specific logging command the provided args
 func (a *journalArgs) copyForBoot(ctx context.Context, w io.Writer, previousBoot int) {
 	if ctx.Err() != nil {
 		return
 	}
+	nativeLoggers, fileLoggers := a.splitNativeVsFileLoggers(ctx)
+	if len(nativeLoggers) > 0 {
+		a.copyServiceLogs(ctx, w, nativeLoggers, previousBoot)
+	}
+
+	// No previous boot logs exists for services that log to a file
+	if len(fileLoggers) > 0 && previousBoot == 0 {
+		copyFileLogs(ctx, w, fileLoggers)
+	}
+
+	// To ensure we get boot logs in the Linux case
+	if len(nativeLoggers) == 0 && len(fileLoggers) == 0 {
+		a.copyServiceLogs(ctx, w, []string{}, previousBoot)
+	}
+}
+
+// splitNativeVsFileLoggers checks if each service logs to native OS logs or to a file and returns a list of services that
+// logs natively vs maybe to a file
+func (a *journalArgs) splitNativeVsFileLoggers(ctx context.Context) ([]string, []string) {
+	var nativeLoggers []string
+	var fileLoggers []string
+
+	for _, service := range a.Services {
+		cmdStr, args := getServiceLogDestinationCmd(service)
+		cmd := exec.Command(cmdStr, args...)
+
+		go func() {
+			<-ctx.Done()
+			if p := cmd.Process; p != nil {
+				p.Kill()
+			}
+		}()
+
+		output, err := cmd.CombinedOutput()
+		if err != nil {
```

Comment:
IIUC this'll only ever be true on Windows as journalctl won't return an error, no? Maybe we could add a comment clarifying what's happening here.